makefile

```diff
@@ -101,30 +102,40 @@ PROJECT := $(APP).$(TARGET)
 # Objects and Paths
 ###############################################################################
 
+APP_INC += -I$(APP_PATH)
+LIB_INC += $(addprefix -I,$(wildcard $(LIB_PATH)/*))
+
 APP_SRC_C += $(wildcard $(APP_PATH)/*.c)
 LIB_SRC_C += $(wildcard $(LIB_PATH)/*/*.c)
 
 APP_SRC_CPP += $(wildcard $(APP_PATH)/*.cpp)
 LIB_SRC_CPP += $(wildcard $(LIB_PATH)/*/*.cpp)
 
-APP_INC += -I$(APP_PATH)
-LIB_INC += $(addprefix -I,$(wildcard $(LIB_PATH)/*))
+UWRT_SRC_C   = $(APP_SRC_C)   $(LIB_SRC_C)
+UWRT_SRC_CPP = $(APP_SRC_CPP) $(LIB_SRC_CPP)
 
-SRC_FILES_C   = $(APP_SRC_C)   $(LIB_SRC_C)
-SRC_FILES_CPP = $(APP_SRC_CPP) $(LIB_SRC_CPP)
+TARGET_SRC += ${TARGET_PATH}/PeripheralPins.c
+TARGET_SRC += ${TARGET_PATH}/system_clock.c
 
 include ${MBED_PATH}/mbedfile
 
-TARGET_OBJ += ${TARGET_PATH}/PeripheralPins.o
-TARGET_OBJ += ${TARGET_PATH}/system_clock.o
+UWRT_C_OBJ := $(subst ../,${OBJ_PATH}/,$(UWRT_SRC_C:.c=.o))
+UWRT_CPP_OBJECTS := $(subst ../,${OBJ_PATH}/,$(UWRT_SRC_CPP:.cpp=.o))
+TARGET_OBJ := $(subst ../,${OBJ_PATH}/,$(TARGET_SRC:.c=.o))
+MBED_OBJ := $(subst ../,${OBJ_PATH}/,$(MBED_OBJ))
+
+OBJECTS += ${UWRT_C_OBJ}
+OBJECTS += ${UWRT_CPP_OBJECTS}
+OBJECTS += ${TARGET_OBJ}
+OBJECTS += ${MBED_OBJ}
 
-OBJECTS += $(SRC_FILES_C:.c=.o) $(SRC_FILES_CPP:.cpp=.o) ${MBED_OBJ} ${TARGET_OBJ}
 
 INCLUDE_PATHS += -I$(CONFIG_PATH)
 INCLUDE_PATHS += -I$(LIB_PATH)
-INCLUDE_PATHS += -I$(MBED_PATH)
-INCLUDE_PATHS += -I${TARGET_PATH}
-INCLUDE_PATHS += $(APP_INC) $(LIB_INC) $(MBED_INC)
+INCLUDE_PATHS += -isystem$(MBED_PATH)
+INCLUDE_PATHS += -isystem$(MBED_INC)
+INCLUDE_PATHS += -isystem${TARGET_PATH}
```

Comment:
What does `-isystem` do?

Reply:
Isystem treats it as a system header. The only really change here is that those headers wont be checked for the warning flags (which they shouldn't be because they're third party code)